mne/source_estimate.py

```diff
@@ -2725,20 +2312,14 @@ def morph_data_precomputed(subject_from, subject_to, stc_from, vertices_to,
     if stc_from.subject is not None and stc_from.subject != subject_from:
         raise ValueError('stc_from.subject and subject_from must match')
 
-    if isinstance(stc_from, VectorSourceEstimate):
-        # Morph the locations of the dipoles, but not their orientation
-        n_verts, _, n_samples = stc_from.data.shape
-        data = morph_mat * stc_from.data.reshape(n_verts, 3 * n_samples)
-        data = data.reshape(morph_mat.shape[0], 3, n_samples)
-        stc_to = VectorSourceEstimate(data, vertices=vertices_to,
-                                      tmin=stc_from.tmin, tstep=stc_from.tstep,
-                                      verbose=stc_from.verbose,
-                                      subject=subject_to)
-    else:
-        data = morph_mat * stc_from.data
-        stc_to = SourceEstimate(data, vertices=vertices_to, tmin=stc_from.tmin,
-                                tstep=stc_from.tstep, verbose=stc_from.verbose,
-                                subject=subject_to)
+    source_morph = SourceMorph(None, subject_from=subject_from,
+                               subject_to=subject_to)
+    source_morph.kind = 'surface'
+    source_morph.data['vertno'] = vertices_to
+    source_morph.data['morph_mat'] = morph_mat
+
+    stc_to = _apply_morph_data(source_morph, stc_from)
```

Comment:
same remark. Could we make SourceMorph class more self sufficient?

Reply:
why not writing

`source_morph(stc_from)`

?

that seems the most natural way for writing it.

regarding populating data with vertices_to and morph_mat I would do it
with at least a private method of SourceMorph class.

Basically you want to isolate as much as possible the morphing logic inside the SourceMorph class.

Reply:
stop reading my mind! :P